plugins/web/opentelemetry-instrumentation-document-load/src/instrumentation.ts

```diff
@@ -141,25 +141,32 @@ export class DocumentLoadInstrumentation extends InstrumentationBase<unknown> {
 
       addSpanPerformancePaintEvents(rootSpan);
 
-      this._endSpan(rootSpan, PTN.LOAD_EVENT_END, entries);
+      this._endSpan(rootSpan, PTN.LOAD_EVENT_END, PTN.FETCH_START, entries);
     });
   }
 
   /**
    * Helper function for ending span
    * @param span
    * @param performanceName name of performance entry for time end
+   * @param startPerformanceName name of the performance entry that has the time start.
    * @param entries
    */
   private _endSpan(
     span: Span | undefined,
     performanceName: string,
+    startPerformanceName: string,
     entries: PerformanceEntries
   ) {
     // span can be undefined when entries are missing the certain performance - the span will not be created
     if (span) {
       if (hasKey(entries, performanceName)) {
-        span.end(entries[performanceName]);
+        // @ts-ignore: Object is possibly 'null'.
+        if(hasKey(entries, startPerformanceName) && entries[startPerformanceName] >  entries[performanceName]) {
+          span.end(entries[startPerformanceName]);
+        } else {
+          span.end(entries[performanceName]);
+        }
```

Comment:
I could not seem to access the span end timestamp here which is why I'm passing the original start value to check against.